default/scripting/ship_hulls/asteroid/SH_ASTEROID.focs.txt

```diff
@@ -29,6 +29,13 @@ Hull
                 OwnedBy empire = Source.Owner
             ]
         ]
+        ResourceSupplyConnected empire = Source.Owner condition = And [
+             Building name = "BLD_SHIPYARD_AST"
+             Or [
+                 OwnedBy empire = Source.Owner
+                 OwnedBy affiliation = AllyOf empire = Source.Owner
+             ]
+        ]
```

Comment:
Don't get the point of this addition: the previous clause requires BLD_SHIPYARD_AST to be in the same system, this new clause requires there be one in supply, the only way the latter can apply without the former is if there's an enemy fleet blockading you: I'd have zero problem with that as a requirement in which case add that as an AND check to the first clause.